pkgs/os-specific/linux/apfs/default.nix

```diff
@@ -44,7 +44,6 @@ stdenv.mkDerivation {
     homepage = "https://github.com/linux-apfs/linux-apfs-rw";
     license = licenses.gpl2Only;
     platforms = platforms.linux;
-    broken = kernel.kernelOlder "4.9";
```

Comment:
Do we actually want to remove this?

- Kernel 4.4 is older than 4.9, but [is supported until 2027](https://en.wikipedia.org/wiki/Linux_kernel_version_history).
- One of the strengths of nixpkgs is being able to combine really old and new software, and if someone who happens to have to run a legacy kernel tries to install APFS with this patch they'd presumably end up with a broken build, and no good explanation why.

Reply:
> Kernel 4.4 is older than 4.9, but is supported until 2027.

There's no official upstream support anymore, though and we don't even have it packaged in here anymore.

> and if someone who happens to have to run a legacy kernel tries to install APFS

In that case you should pin nixpkgs to a revision that's known to work. It would be really bad if we could never clean up anything in this repository.

Reply:
> > Kernel 4.4 is older than 4.9, but is supported until 2027.
> 
> There's no official upstream support anymore, though

No, there _is_ upstream support according to the linked page.

> > and if someone who happens to have to run a legacy kernel tries to install APFS
> 
> In that case you should pin nixpkgs to a revision that's known to work.

People regularly have to combine old and new software; nixpkgs is sort of famous ([Searching and installing old versions of Nix packages](https://lazamar.github.io/download-specific-package-version-with-nix/), [Building 15-year-old software with Nix](https://blinry.org/nix-time-travel/)) for supporting just this use case.

> It would be really bad if we could never clean up anything in this repository.

That's the curse of being backwards compatible. Although "never" in this case could just mean 2027.

Reply:
>  see a group called the "Civil Infrastructure Platform" is providing some limited ongoing maintenance. We have never packaged or claimed to support CIP-maintained kernels, and I don't think that organisations that need software maintained on the 25-50-year timescale they strive for are likely to find Nixpkgs (where support is guaranteed for, at most, seven months at a time) very useful.


Apart from @alyssais' remark we even drop kernels that are not supported by _upstream_ anymore (i.e. 4.4 doesn't even exist in nixpkgs anymore) @l0b0 .

> People regularly have to combine old and new software; nixpkgs is sort of famous (Searching and installing old versions of Nix packages, Building 15-year-old software with Nix) for supporting just this use case.

This is _precisely_ what I suggested, i.e. using old nixpkgs revs and ideally and/or pinning nixpkgs down to an older rev.